packages/dds/tree/src/tableSchema.ts

```diff
@@ -831,7 +835,12 @@ export namespace System_TableSchema {
 			/* TInsertable */ object & TableInsertableType,
 			/* ImplicitlyConstructable */ true,
 			/* Info */ typeof tableFields
-		> = Table;
+		> & {
+			/**
+			 * Create an empty table.
+			 */
+			empty(): TableValueType;
```

Comment:
I don't think this is correct. We rely on some really subtle details to make subclassing work properly. TableValueType is the instance type of the class we are returning. Its impossible to refer to the instance type of the derived class (it doesn't exist yet) but it is what empty must actually return. This works for the constructor since subclassing updates the constructor to return the correct type, then the node type parameter is re-inferred from that to be the derived type when using it as a TreeNodeSchemaClass. This does not work for static factory methods, which is why they they need to generically capture the `this` type.


I think with the differing types, the assignment here only compiles due to typescript not type checking upcasts of generic this robustly.

Reply:
Sure enough. The test case you recommended confirms this.